torch/csrc/jit/python/init.cpp

```diff
@@ -400,19 +400,44 @@ void initJITBindings(PyObject* module) {
       .def(
           "_jit_pass_onnx_unpack_quantized_weights",
           [](std::shared_ptr<Graph>& graph,
-             std::map<std::string, at::Tensor>& paramsDict) {
+             std::map<std::string, IValue>& paramsDict) {
             UnpackQuantizedWeights(graph, paramsDict);
             return paramsDict;
           },
           pybind11::return_value_policy::move)
       .def(
           "_jit_pass_onnx_quantization_insert_permutes",
           [](std::shared_ptr<Graph>& graph,
-             std::map<std::string, at::Tensor>& paramsDict) {
+             std::map<std::string, IValue>& paramsDict) {
             insertPermutes(graph, paramsDict);
             return paramsDict;
-          },
-          pybind11::return_value_policy::move);
+          })
+      .def(
+          "_jit_pass_filter_non_tensor_arguments",
+          [](std::map<std::string, IValue> params) {
+            std::map<std::string, at::Tensor> retval;
+            for (auto& kv : params) {
+              if (kv.second.isTensor()) {
+                retval[std::move(kv.first)] = std::move(kv.second).toTensor();
+              }
+            }
+            return retval;
+          })
+      .def("_jit_decay_packed_param_input_types", [](Graph& g) {
+        for (Value* i : g.inputs()) {
```

Comment:
why is it a separate pass? can't `unpackQuantizedWeightsHelper ` handle it? it logically belongs there (when we replace class instance with tensor)

Reply:
I don't really follow where you're referring to when you say `unpackQuantizedWeightsHelper` should handle it

Reply:
I meant that unpackQuantizedWeightsHelper does the modification of params. So it's logical for it to also set the graph input to be of the correct type (or maybe delete the input all together). Otherwise these two passes (_jit_pass_filter_non_tensor_arguments and _jit_decay_packed_param_input_types) are a bit magical to be called from the top level

Reply:
It doesn't fit into the existing infrastructure:
```
======================================================================
ERROR: test_torchbind_lower_graph (__main__.TestScript)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_jit.py", line 4630, in test_torchbind_lower_graph
    onnx_model = export_to_onnx(model, x, input_names)
  File "test/test_jit.py", line 4624, in export_to_onnx
    operator_export_type=torch.onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK)
  File "/data/users/jamesreed/pytorch/torch/onnx/__init__.py", line 172, in export
    custom_opsets, enable_onnx_checker, use_external_data_format)
  File "/data/users/jamesreed/pytorch/torch/onnx/utils.py", line 92, in export
    use_external_data_format=use_external_data_format)
  File "/data/users/jamesreed/pytorch/torch/onnx/utils.py", line 519, in _export
    fixed_batch_size=fixed_batch_size)
  File "/data/users/jamesreed/pytorch/torch/onnx/utils.py", line 340, in _model_to_graph
    in_vars, in_desc = torch.jit._flatten(tuple(args) + tuple(params))
RuntimeError: Only tuples, lists and Variables supported as JIT inputs/outputs. Dictionaries and strings are also accepted but their usage is not recommended. But got unsupported type torch._C.ScriptObject
```

And I don't want to rabbit-hole down another fractal of redesign when this solution just works